apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-global-styles-click.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './track-record-event';
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_global_styles_click`.
+ *
+ * @returns {{handler: Function, selector: string, type: string}} event object definition.
+ */
+export default () => ( {
+	selector: `.edit-site-header__actions button[aria-label="${ __( 'Global Styles' ) }"]`,
```

Comment:
I wonder if there is a data store selector that could help track this so it isn't reliant on click events.

Reply:
There is a data store selector for that 😄 Question is what we want to track exactly. Since Global Styles sidebar can be opened by two different buttons: one in the header, and one in the more menu. 

The spreadsheet stated "track click in header". If we just want to fire the event whenever the global styles sidebar is toggled then we definitely need to use the data store for that.

Reply:
Tried to implement it by intercepting the redux action `enableComplementaryArea` and `disableComplementaryArea`. Here are my findings:
* We can use `enableComplementaryArea` to track "global styles sidebar open" action.
* We can use `disableComplementaryArea` to track "global styles sidebar close" action. However, this will track all the possible ways too, which means: button in the header, button in the more menu **and X button in the sidebar**. This won't track closes triggered by opening other sidebars, for example opening the settings sidebar won't trigger this one, which is good.
* We can differentiate between which button opened the global styles sidebar. `header` vs `more-menu`
* We can differentiate between which button closed the global styles sidebar. We have 3 buttons, but we can only differentiate between `header or sidebar X` vs `more-menu`. (Checking for `more-menu` existence)


After all, using the redux action would be preferred way, but it comes with its own limitations. Though I don't think it's that bad, so maybe we should give it a try. What do you think?

Reply:
> The spreadsheet stated "track click in header". If we just want to fire the event whenever the global styles sidebar is toggled then we definitely need to use the data store for that.

I think that event in the spreadsheet is next to the current existing events for dotcom's legacy FSE Global Styles, which I think the only way to open is by clicking the button.

> After all, using the redux action would be preferred way, but it comes with its own limitations. Though I don't think it's that bad, so maybe we should give it a try. What do you think?

It seems like a promising way to go.  Im not sure whether or not we need to track when the bar is closed, hopefully the upcoming discussion on the p2 I linked will help clarify that as well.

Reply:
After the discussion in the p2 (paYE8P-V2-p2#comment-821), I don't think we really need to track the close action.

What we specifically want is a `wpcom_block_editor_global_styles_tab_selected` event, with a `tab` property logging the `root` or `blockType` tab.

Now, apparently the GS tabs are handled internally by the `TabPanel` component, so we cannot infer them with Redux.
On the other hand, this means that `enableComplementaryArea` would _always_ open GS to the `root` tab.

What we could do is record the same event in two different ways:
- On `enableComplementaryArea` with `{ scope: 'core/edit-site', item: 'edit-site/global-styles' }`:
  - Track `wpcom_block_editor_global_styles_tab_selected` with `{ tab: 'root', open: true }`.
  - The `open` property is not strictly needed, but I think... why not? 😄
- When switching tabs in the GS sidebar (`.edit-site-global-styles-sidebar .components-tab-panel__tabs`):
  -  Track `wpcom_block_editor_global_styles_tab_selected` with `{ tab: 'root' }` or `{ tab: 'blockType' }`.

I'm not quite sure if we have an established way to track tab changes without specific class names, but I guess we might be able to target through first and last child pseudo classes.